lib/ok_computer/built_in_checks/active_record_migrations_check.rb

```diff
@@ -2,36 +2,33 @@ module OkComputer
   class ActiveRecordMigrationsCheck < Check
     # Public: Check if migrations are pending or not
     def check
-      return unsupported unless supported?
+      # this check is only valid if config.active_record.migration_error
+      # is set to false rather than :page_load
+      # :page_load is the default in development, otherwise this is false
+      if Rails.configuration.active_record.migration_error
+        return mark_message("NO pending migrations") if Rails.env.development?
 
-      if needs_migration?
-        mark_failure
-        mark_message "Pending migrations"
-      else
-        mark_message "NO pending migrations"
+        return check_on_page_load
       end
-    end
 
-    def needs_migration?
-      if ActiveRecord::Migrator.respond_to?(:needs_migration?) # Rails <= 5.1
-        ActiveRecord::Migrator.needs_migration?
-      else # Rails >= 5.2
-        ActiveRecord::Base.connection.migration_context.needs_migration?
+      if Gem::Version.new(Rails.version) >= Gem::Version.new("7.1")
+        ActiveRecord::Migration.check_all_pending!
+      else
+        ActiveRecord::Migration.check_pending!
       end
-    end
 
-    def supported?
-      ActiveRecord::Migrator.respond_to?(:needs_migration?) ||
-        (ActiveRecord::Base.connection.respond_to?(:migration_context) &&
-         ActiveRecord::Base.connection.migration_context.respond_to?(:needs_migration?))
+      mark_message "NO pending migrations"
+    rescue ActiveRecord::PendingMigrationError
+      mark_failure
+      mark_message "Pending migrations"
```

Comment:
[NIT] I think this logic would read better if it were put in a `#needs_migartion?` method.  i.e. Keep a similar signature as the prior implementation, but just change the `#needs_migration?` method.

Reply:
refactored methods to improve clarity